### PR TITLE
postgres

### PR DESCRIPTION
**Pull request by dmgaldi**
Sorry, this is a beefy PR.

Let me know if you want to look through it together again.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -42,6 +42,8 @@ RUN stat /workspace/build/mbio-site-artifact.tar.gz
 RUN cp -r project_home/OAuth2Server .
 RUN cd /workspace/OAuth2Server && bash EuPathDB/bin/build.sh WEB-INF/OAuthConfig.json
 
+RUN echo "stub"
```
Comment (ryanrdoherty) [inline Dockerfile]: Remove

File: bin/configureOauth.sh [hunk 2]
```diff
@@ -51,22 +53,25 @@ jq \
...
   . |
   (.allowedClients[] | select(.clientId == $wdk_client_id) |  .clientSecrets) = [$wdk_client_secret] |
   (.allowedClients[] | select(.clientId == $wdk_client_id) |  .clientDomains) = ["*.amoebadb.org","*.apidb.org","*.cryptodb.org","*.eupathdb.org","*.fungidb.org","*.giardiadb.org","*.hostdb.org","*.microsporidiadb.org","*.orthomcl.org","*.piroplasmadb.org","*.plasmodb.org","*.schistodb.net","*.toxodb.org","*.trichdb.org","*.tritrypdb.org","*.clinepidb.org","*.microbiomedb.org","*.vectorbase.org","*.veupathdb.org"] |
   (.allowedClients[] | select(.clientId == $service_client_id) |  .clientSecrets) = [$service_client_secret] |
   (.allowedClients[] | select(.clientId == $service_client_id) |  .clientDomains) = ["*.amoebadb.org","*.apidb.org","*.cryptodb.org","*.eupathdb.org","*.fungidb.org","*.giardiadb.org","*.hostdb.org","*.microsporidiadb.org","*.orthomcl.org","*.piroplasmadb.org","*.plasmodb.org","*.schistodb.net","*.toxodb.org","*.trichdb.org","*.tritrypdb.org","*.clinepidb.org","*.microbiomedb.org","*.vectorbase.org","*.veupathdb.org"] |
-
+  (.allowedClients[] | select(.clientId == $service_client_id) |  .clientSecrets) = [$service_client_secret] |
```
Comment (ryanrdoherty) [inline bin/configureOauth.sh]: Is this not a replication of line 61?

File: bin/configureOauth.sh [hunk 3]
```diff
@@ -51,22 +53,25 @@ jq \
...
   .authenticatorConfig.login = $db_login | 
   .authenticatorConfig.password = $db_password |
-  .authenticatorConfig.connectionUrl = $db_url
+  .authenticatorConfig.connectionUrl = $db_url |
+  .authenticatorConfig.platform = $oauth_db_platform
 ' \
 $unmodifiedConfig \
 > "WEB-INF/OAuthConfig.json"
 
+rm -f WEB-INF/OAuthSampleConfig.json
```
Comment (ryanrdoherty) [inline bin/configureOauth.sh:73]: Not sure this does any harm but if we're going to remove it, might as well remove the Mbio one too.

File: docker-compose.yml [hunk 4]
```diff
@@ -9,7 +9,7 @@ networks:
     #
     # If rabbit is not local then the network is not defined externally and we
     # need to define it ourselves (set this to false)
-    external: ${RABBITMQ_IS_LOCAL:-false}
+#    external: ${RABBITMQ_IS_LOCAL:-false}
```
Comment (ryanrdoherty) [inline docker-compose.yml]: Did you have trouble with this?  I think I set to true in base env and it worked fine.

File: docker-compose.yml [hunk 5]
```diff
@@ -55,6 +55,7 @@ services:
       WDK_CACERTS_FILE: ${SECRETS_MOUNT:-/secrets}/${WDK_CACERTS_FILENAME:-cacerts}
       OAUTH_SIGNING_KEY_STORE: ${SECRETS_MOUNT:-/secrets}/${OAUTH_SIGNING_KEY_STORE_FILENAME:-oauth-keys.pkcs12}
       OAUTH_SIGNING_KEY_STORE_PW: ${OAUTH_SIGNING_KEY_STORE_PW}
+      OAUTH_DB_PLATFORM: PostgreSQL
```
Comment (ryanrdoherty) [inline docker-compose.yml]: Change to config value

File: docker-compose.yml [hunk 6]
```diff
@@ -82,6 +83,9 @@ services:
       ACCT_DB_PASSWORD: ${ACCT_DB_PASSWORD}
       ACCT_DB_CONNECTION_URL: ${ACCT_DB_CONNECTION_URL}
 
+    depends_on:
```
Comment (ryanrdoherty) [inline docker-compose.yml]: Move to pg-specific compose file

File: docker-compose.yml [hunk 7]
```diff
@@ -98,6 +102,37 @@ services:
       - "traefik.http.routers.${TRAEFIK_MBIOWDK_ROUTER:-mbiowdk-dev}.entrypoints=${TRAEFIK_ENTRYPOINTS:-local}"
       - "traefik.docker.network=traefik"
 
+  #####################
```
Comment (ryanrdoherty) [inline docker-compose.yml]: Move to pg-specific compose file

File: docker-compose.yml [hunk 8]
```diff
@@ -382,22 +421,14 @@ services:
     image: veupathdb/vdi-service:${VDI_SERVICE_TAG:-latest}
     depends_on:
     - cache-db
+    - edapostgres
```
Comment (ryanrdoherty) [inline docker-compose.yml]: I think you can add these depends_on records in the pg compose.

File: docker-compose.yml [hunk 9]
```diff
@@ -430,10 +461,14 @@ services:
       USER_UPLOAD_QUOTA: ${USER_UPLOAD_QUOTA}
 
       # Account DB connection details
-      ACCT_DB_TNS_NAME: ${ACCT_DB_TNS_NAME:?}
-      ACCT_DB_USER: ${ACCT_DB_LOGIN:?}
-      ACCT_DB_PASS: ${ACCT_DB_PASSWORD:?}
-      ACCT_DB_POOL_SIZE: ${ACCT_DB_POOL_SIZE:-5}
+#      ACCT_DB_TNS_NAME: ${ACCT_DB_TNS_NAME:?}
```
Comment (ryanrdoherty) [inline docker-compose.yml]: If we make this default an empty string, does it still pick up the URL parts below?  Would be nice to be able to configure both ora and pg with these values.  Or maybe it's time to try your fix where we can use the url parts with oracle?

File: docker-compose.yml [hunk 10]
```diff
@@ -576,136 +591,23 @@ services:
...
-      DB_CONNECTION_LDAP_CLINEPI: ${DB_CONNECTION_LDAP_CLINEPI:?}
-      DB_CONNECTION_PASS_CLINEPI: ${DB_CONNECTION_PASS_CLINEPI:?}
-      DB_CONNECTION_DATA_SCHEMA_CLINEPI: ${DB_CONNECTION_DATA_SCHEMA_CLINEPI:?}
-      DB_CONNECTION_CONTROL_SCHEMA_CLINEPI: ${DB_CONNECTION_CONTROL_SCHEMA_CLINEPI:?}
-      DB_CONNECTION_POOL_SIZE_CLINEPI: ${DB_CONNECTION_POOL_SIZE_CLINEPI}
-
       DB_CONNECTION_ENABLED_MICROBIOME: ${DB_CONNECTION_ENABLED_MICROBIOME}
       DB_CONNECTION_NAME_MICROBIOME: ${DB_CONNECTION_NAME_MICROBIOME:?}
-      DB_CONNECTION_LDAP_MICROBIOME: ${DB_CONNECTION_LDAP_MICROBIOME:?}
+      DB_CONNECTION_LDAP_MICROBIOME: ${DB_CONNECTION_LDAP_MICROBIOME}
```
Comment (ryanrdoherty) [inline docker-compose.yml]: Oh, does ? make it required?  I was thinking absence of a default made it choke if no value was supplied.

**Review by ryanrdoherty — CHANGES_REQUESTED** [review]
Mostly just making sure we keep oracle compatibility.

File: bin/configureOauth.sh (hunk 3 above)
Comment (dmgaldi) [reply bin/configureOauth.sh:73]: Oops, this was added when I was confused about an issue I was having. I'll remove it

File: docker-compose.yml (hunk 4 above)
Comment (dmgaldi) [reply docker-compose.yml]: I did, but I probably have it set wrong in my .env. I don't think we ever expect rabbit to be external, so maybe I'll just remove it.

File: bin/configureOauth.sh (hunk 2 above)
Comment (dmgaldi) [reply bin/configureOauth.sh]: Bad copy-paste that got left in, thanks

File: docker-compose.yml (hunk 9 above)
Comment (dmgaldi) [reply docker-compose.yml]: It does work to have this as optional, being an empty string if it's omitted. It's unused if PostgreSQL is specified as the platform.

File: docker-compose.yml (hunk 10 above)
Comment (dmgaldi) [reply docker-compose.yml]: Yeah, I think it defaults to an empty string if it's unspecified.

File: docker-compose.dev.yml [hunk 11]
```diff
@@ -58,7 +58,7 @@ services:
     - rabbitmq
     volumes:
       - type: bind
-        source: ${VDI_MINIO_DATA_DIR:-/var/www/Common/persistentData/minio}
+        source: ${VDI_MINIO_DATA_DIR:-$LOCAL_MINIO_DIR}
```
Comment (ryanrdoherty) [inline docker-compose.dev.yml:61]: A little off topic but in my work to fix up config, I'd like to have a "global" "siteDataDirectory" which would have a default value of /var/www/Common.  But everything else would live under that.  Do you think "/persistentData/minio" is a good place?  Would also like to have "/persistentData/postgres".  In the current model we're planning a single container with multiple pg databases which may each contain multiple schemas, yes?  If a single container/instance, then naming it "postgres" is probably fine- I don't think we need to worry about it being confused with the compute or VDI internal postgres's.

File: files/gus_home/config/MicrobiomeDB/model-config.xml.tmpl [hunk 12]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <modelConfig modelName="microbiomeModel"
              webServiceUrl="local"
-             smtpServer="127.0.0.1"
+             smtpServer="%%SMTP_HOST%%"
```
Comment (ryanrdoherty) [inline files/gus_home/config/MicrobiomeDB/model-config.xml.tmpl:4]: Need to add new username/password fields here with envs and docker compose too, with empty values as defaults.

**Review by ryanrdoherty — APPROVED** [review]
A few more fixes and thoughts.

File: docker-compose.dev.yml (hunk 11 above)
Comment (dmgaldi) [reply docker-compose.dev.yml:61]: Yeah, that works! Do you want me to update in this PR, or have you already started factoring this?

**Pull request merged**